libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Recognizers/CrossTrainedRecognizerSet.cs

```diff
@@ -0,0 +1,221 @@
+            }
+
+            // this is the consensusRecognizer to use
+            string consensusRecognizerId = null;
+            foreach (var recognizer in Recognizers)
+            {
+                var intent = intents[recognizer.Id];
+
+                if (!IsRedirect(intent))
+                {
+                    // we have a real intent and it's the first one we found.
+                    if (consensusRecognizerId == null)
+                    {
+                        consensusRecognizerId = recognizer.Id;
+                    }
+                    else
+                    {
+                        // we have a second recognizer with an intent
+
+                        // if one of them is None intent, then go with the other one.
+                        if (intent == NoneIntent)
+                        {
+                            // then we are fine with the one we have, just ignore this one
+                            continue;
+                        }
+                        else if (intents[consensusRecognizerId] == "None")
+                        {
+                            // then we can drop the old one and go with the new one instead
+                            consensusRecognizerId = recognizer.Id;
+                        }
+                        else
+                        {
+                            // ambigious because of 2 real intents, and neither are None so return AmbigiousIntent
+                            return CreateAmbigiousIntentResult(text, recognizerResults);
+                        }
+                    }
+                }
+                else
+                {
+                    // get the redirectId and redirectIntent 
+                    var redirectId = GetRedirectId(intent);
+                    var redirectIntent = intents[redirectId];
+
+                    // if the redirectIntent is itself a redirect, then we have double redirect which means disagreement.
+                    if (IsRedirect(redirectIntent))
+                    {
+                        // we have ambiguity, return AmbigiousIntent
+                        return CreateAmbigiousIntentResult(text, recognizerResults);
+                    }
+                }
+            }
+
+            // we have consensus for consensusRecognizer, return the results of that recognizer as the result.
+            return recognizerResults[consensusRecognizerId];
+        }
+
+        private RecognizerResult CreateAmbigiousIntentResult(string text, Dictionary<string, RecognizerResult> recognizerResults)
+        {
+            // create IntentScore with { "recognizerId" : { ...RecognizerResult.. } }
+            var ambigiousScore = new IntentScore()
+            {
+                Score = 0.5F
+            };
+
+            foreach (var result in recognizerResults)
+            {
+                ambigiousScore.Properties[result.Key] = result.Value;
+            }
```

Comment:
Wouldn't it make more sense to filter out everyone that thinks someone else has the answer?  If A->B and B says I got it and C say I got it, shouldn't you have B & C and not A?